### PR TITLE
Add setup.py to provide the ability to install as a module

### PR DESCRIPTION
**Pull request by elebertus**
Hey!

You already had the foresight to write this mostly as a module first. I figured that instead of adding more modules at least for the time being we could just add some simple functionality to install the module.

This also has the added bonus of being able to forego `python -m updater` and simply add a `wow-addon-updater` command to the user's path.

Edit: I credited you and set the url to the repo. As far as publishing this goes or making any other cosmetic changes. I'm also happy to make any changes and push them.

Quick example of what this provides:
```bash
# install the module
(master-wow-addon-updater) $: python setup.py install
<snip>
# the imports in __init__.py make using this as a library feel more clean
# updater.site.__init__ was left untouched since that's more of an "internal" module and set of classes
(master-wow-addon-updater) $: python -c 'from updater.manager import AddonManager;print(AddonManager.__module__)'
updater.manager.addon_manager
# using the entry_points from setup.py instead of python -m updater
(master-wow-addon-updater) $: wow-addon-updater
Installing/updating addon: details to version: DetailsRetail.8.3.0.7356.141...



Name                          Prev. Version                 New Version
────                          ─────────────                 ───────────
details                       -----                         DetailsRetail.8.3.0.7356.141
(master-wow-addon-updater) $: cat in.txt
https://www.curseforge.com/wow/addons/details
(master-wow-addon-updater) $: ls $(grep 'WoW Addon Location' config.ini | awk -F'=' '{print $NF}')
Details  Details_DataStorage  Details_EncounterDetails  Details_RaidCheck  Details_Streamer  Details_TinyThreat  Details_Vanguard
```

### CODE REVIEW

**Comment by elebertus** [comment]
@grrttedwards ^^ I didn't add `Pipfile.lock` since there have been some changes to other deps, I'm happy to add that as well if you like.

**Comment by grrttedwards** [comment]
Hello, hello, hello!

Wow, I glanced over this and it looks like a great improvement. I’ll take a closer look as soon as I can. At this point in the project, I would prefer to make no breaking changes for existing users. I think that’s the case here, so that’s awesome.

**Comment by elebertus** [comment]
Awesome! Let me know if there are any changes or questions :+1:

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+setup(
+    name="wow_addon_updater",
+    version="v1.7.1",
+    description=(
+        "This utility provides an alternative to the Twitch/Curse "
+        "client for management and updating of addons for World of Warcraft. "
+    ),
+    author="Garrett Edwards",
+    url="https://github.com/grrttedwards/wow-addon-manager",
```
Comment (grrttedwards) [inline setup.py:12]:
```suggestion
    url="https://github.com/grrttedwards/wow-addon-manager",
    license="GPLv3",
    classifiers=[
        "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
        "Programming Language :: Python :: 3 :: Only",
        "Operating System :: OS Independent",
    ],
```

I built the ... package? wheel? egg? whatever it's called 😄 and noticed a few little holes in the metadata. There's also a
```
Description: UNKNOWN
Platform: UNKNOWN
```
that goes into the `PKG-INFO` which we may want to fill in.

File: README.md [hunk 2]
```diff
@@ -151,6 +169,11 @@ pipenv run coverage run --source=updater -m unittest -v
 pipenv run coverage report
 ```
 
+Updating deps in setup.py:
+```bash
+pipenv-setup sync -d
```
Comment (grrttedwards) [inline README.md:174]: Love this! Was thinking to myself "ugh will I have to manually sync the locked dependencies?!" this module rules.

File: README.md [hunk 3]
```diff
@@ -51,6 +51,24 @@ To run directly from the command line, use `pipenv run`:
...
+pipenv install
+pipenv shell
+# Make sure you're in a virtual environment! pipenv shell should place you in the venv
+# created by `pipenv install` but to be sure you can check with `which python` and
+# verify that it's $VIRTUAL_ENV/bin/python
+
+# Using the distutils method
+python setup.py install
+# Using pip
+pip install .
```
Comment (grrttedwards) [inline README.md:65]: I'll admit I was blindly spitting these lines into my terminal, and hesitated when I ran `python setup.py install` and `pip install .` back to back and it failed. Maybe clarify this instruction with something like

```suggestion
# Either using the distutils method
python setup.py install
# Or using pip
pip install .
```

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        "certifi==2020.6.20",
+        "chardet==3.0.4",
+        "cloudscraper==1.2.46",
+        "idna==2.10; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+        "pyparsing==2.4.7; python_version >= '2.6' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+        "requests==2.24.0",
+        "requests-toolbelt==0.9.1",
+        "soupsieve==2.0.1; python_version >= '3.5'",
+        "urllib3==1.25.10; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4' and python_version < '4'",
+    ],
```
Comment (grrttedwards) [inline setup.py:62]: A little concerned about what this actually generated. On some platforms, I've run into issues with the SSL version used when communicating with C*oudflare (trying to remember, I think the culprit was Windows), so there's a soft requirement on Python 3.8 there. Do the lines like `python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4' and python_version < '4'"` enable too loose of a version requirement? Is there a way to require a Python version for the package consumer?

File: .gitignore [hunk 5]
```diff
@@ -1,5 +1,6 @@
 # Addon input files
```
Comment (grrttedwards) [inline .gitignore:1]: Can you tack on 
```
build/ 
dist/ 
wow_addon_updater.egg-info/ 
```
at the end? Like to keep it clean, and avoids people committing some junk accidentally.

File: setup.py (hunk 4 above)
Comment (grrttedwards) [reply setup.py:62]: Maybe with `python_requires='>=3.8.0'` in `setup.py`?

File: README.md (hunk 2 above)
Comment (elebertus) [reply README.md:174]: I'll be honest I hadn't worked much with pipenv or Pipfiles and once I thought "oh right, how do I sync these" I got really happy that problem was solved haha

File: setup.py (hunk 4 above)
Comment (elebertus) [reply setup.py:68]: Exactly your second comment. [python_requires](https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires) allows for exactly that, or you can technically do the same thing (with different strings) as above. Just `python_requires='>=3.8.0'`.

3.6.x is still fairly common, but as you're saying if it really is a hard requirement due to other deps 3.8 is fine and readily available.

I can toss that in for sure.

File: .gitignore (hunk 5 above)
Comment (elebertus) [reply .gitignore:1]: :+1:

**Comment by elebertus** [comment]
I think I covered everything, give it a double check when you have time :+1: 

Also in terms of publishing this as a module, that's entirely up to you. It is a way to get more exposure to your project, users, developers, etc. Which can also mean just more testing and bug reporting, or features. Completely up to you!

**Comment by grrttedwards** [comment]
Looks good. Thanks for the follow up -- and sorry this took 22 days!  I changed the base branch to `develop`, since I guess I started using gitflow recently on this project 😛

**Pull request merged**